src/bin/wc.rs

```diff
@@ -103,24 +103,66 @@ struct Flags {
     count_bytes: bool,
 }
 
+fn u64_num_digits(val: u64) -> usize {
+    if val == 0 {
+        1
+    } else {
+        (val as f32).log10() as usize + 1
+    }
+}
+
 impl Flags {
-    fn print_count<'a, W: Write>(self, count: Counter, file: &'a str, stdout: &mut W, stderr: &mut Stderr) {
-        stdout.write(b"    ").try(&mut *stderr);
+    fn print_counts<W: Write>(self, mut counts: Vec<(Counter, String)>, stdout: &mut W, stderr: &mut Stderr) {
```

Comment:
You should add the `'a` back and use it in the string, making it a string slice. That would remove all those extra allocations, without sacrificing anything else.